scripts/ci/source_code_static_analysis.py

```diff
@@ -0,0 +1,154 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+
+import os
+import sys
+import multiprocessing
+
+from pylint import lint
+from flake8.main import application
+
+from util import get_repo_root
+
+# import argparse
+# from verify_codeowners import main as _verify_codeowners
+# from verify_license import main as _verify_license
+
+
+SDK_FILE_MARKER = r"# Code generated by Microsoft (R) AutoRest Code Generator."
+
+
+# build list of sdk files
+def _get_sdk_module_list(root_dir):
+    dir_paths = [os.path.join(root_dir, path) for path in os.listdir(root_dir)]
+    sdk_files = []
+
+    for path in dir_paths:
+        if os.path.isdir(path):
+            sdk_files.extend(_get_sdk_module_list(path))
+        elif _is_sdk_file(path):
+            sdk_files.append(os.path.dirname(path))
+
+    return sdk_files
+
+
+# check if the current file is a python sdk file
+def _is_sdk_file(file_path):
+    # don't bother opening non-python files. e.g pyc files.
+    if not file_path.endswith(".py"):
+        return False
+    with open(file_path, encoding='utf-8') as f:
+        for line in f:
+            if SDK_FILE_MARKER in line:
+                return True
+    return False
+
+
+# build list of modules that start with prefix "azext_" in the src
+def _get_azext_module_paths(root_dir):
+    src_root = os.path.join(root_dir, "src")
+    src_paths = [os.path.join(src_root, path) for path in os.listdir(src_root)]
+    ext_modules = []
+    for ext_root in src_paths:
+        if os.path.isdir(ext_root):
+            paths = [path for path in os.listdir(ext_root)]
+            ext_modules.extend([os.path.join(ext_root, path) for path in paths if path.startswith("azext_")])
+    return ext_modules
+
+
+# build list of python ci scripts
+def _get_ci_py_file_paths(directory):
+    return [os.path.join(directory, path) for path in os.listdir(directory) if path.endswith(".py")]
+
+
+def _run_pylint(module_paths, ignored_modules=None, rcfile=None, cpu_count=1):
+    pylint_opts = []
+
+    if ignored_modules:
+        pylint_opts.append("--ignore={}".format(ignored_modules))
+    if rcfile:
+        pylint_opts.append("--rcfile={}".format(rcfile))
+
+    pylint_opts.append("--jobs={}".format(cpu_count))
+    pylint_opts.extend(module_paths)
+
+    try:
+        lint.Run(pylint_opts)
+    except SystemExit as se:
+        # 0:  everything is fine
+        # 1:  Fatal message issued
+        # 2:  Error message issued
+        # 4:  Warning message issued
+        # 8:  Refactor message issued
+        # 16: Convention message issued
+        # 32: Usage error
+        if se.code not in [0, 8, 16, 24]:
```

Comment:
It is to add 24 by design. Right? Because I just don't see the help message for 24. Just a double check. 😄

Reply:
Yes. 24 = 8 + 16. Sometimes Refactor and Convention message will occur in the mean time.